Spigot-Server-Patches/0580-Only-consider-chunks-Loaded-if-at-BORDER-status.patch

```diff
@@ -0,0 +1,158 @@
+     }
+     // CraftBukkit end
+ 
+diff --git a/src/main/java/net/minecraft/server/ChunkProviderServer.java b/src/main/java/net/minecraft/server/ChunkProviderServer.java
+index 45c142c227bc258bbd5c8c03c6012663dccf8f3f..d119d68bb69baecdd7b896e6efb41251fa226adb 100644
+--- a/src/main/java/net/minecraft/server/ChunkProviderServer.java
++++ b/src/main/java/net/minecraft/server/ChunkProviderServer.java
+@@ -19,6 +19,7 @@ import java.util.function.Supplier;
+ import javax.annotation.Nullable;
+ import com.destroystokyo.paper.exception.ServerInternalException;
+ import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap; // Paper
++import it.unimi.dsi.fastutil.longs.LongIterator; // Paper
+ import org.apache.logging.log4j.LogManager;
+ import org.apache.logging.log4j.Logger;
+ 
+@@ -822,7 +823,9 @@ public class ChunkProviderServer extends IChunkProvider {
+             //List<PlayerChunk> list = Lists.newArrayList(this.playerChunkMap.f()); // Paper
+             //Collections.shuffle(list); // Paper
+             // Paper - moved up
+-            final int[] chunksTicked = {0}; this.playerChunkMap.forEachVisibleChunk((playerchunk) -> { // Paper - safe iterator incase chunk loads, also no wrapping
++            final int[] chunksTicked = {0};
++            for (LongIterator iterator = this.playerChunkMap.loadedChunks.iterator() ; iterator.hasNext() ; ) { // Paper - iterate only loaded chunks
++                PlayerChunk playerchunk = this.playerChunkMap.getVisibleChunk(iterator.nextLong());// Paper - iterate only loaded chunks
+                 Optional<Chunk> optional = ((Either) playerchunk.a().getNow(PlayerChunk.UNLOADED_CHUNK)).left();
+ 
+                 if (optional.isPresent()) {
+@@ -850,7 +853,7 @@ public class ChunkProviderServer extends IChunkProvider {
+                         }
+                     }
+                 }
+-            });
++            };// Paper - use for instead of forEachVisibleChunk
+             this.world.getMethodProfiler().enter("customSpawners");
+             if (flag1) {
+                 try (co.aikar.timings.Timing ignored = this.world.timings.miscMobSpawning.startTiming()) { // Paper - timings
+diff --git a/src/main/java/net/minecraft/server/PlayerChunkMap.java b/src/main/java/net/minecraft/server/PlayerChunkMap.java
+index fcd3388d88aad6e897b73e2c823267ada327cbda..3f7da1a74eb881ddc30b20143c20fe162543c20f 100644
+--- a/src/main/java/net/minecraft/server/PlayerChunkMap.java
++++ b/src/main/java/net/minecraft/server/PlayerChunkMap.java
+@@ -1021,16 +1021,18 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
+                 this.a(i, playerchunk);
+             } else {
+                 if (this.pendingUnload.remove(i, playerchunk) && ichunkaccess != null) {
++                    // Paper start - coment out and move to ChunkUnloadEvent
+                     if (ichunkaccess instanceof Chunk) {
+-                        ((Chunk) ichunkaccess).setLoaded(false);
```

Comment:
Empty if statement

Reply:
we try to keep diff minimal, if they add something else here, it may then be needed to keep the chunk check.